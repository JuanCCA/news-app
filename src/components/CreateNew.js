import React, {useState} from 'react'
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import FormCreateNew from './FormCreateNew';

const CreateNew = () => {

    const [show, setShow] = useState(false);

    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
  

    return (
        <>
          <Button variant="primary" onClick={handleShow}>
           Create New
          </Button>
    
          <Modal show={show} onHide={handleClose}>
            <Modal.Header closeButton>
              <Modal.Title>Create a New</Modal.Title>
            </Modal.Header>
            <Modal.Body>

                <FormCreateNew />
                
            </Modal.Body>

          </Modal>
        </>
      );
}

export default CreateNew
