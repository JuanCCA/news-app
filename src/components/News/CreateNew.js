import React, {useState} from 'react'
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import FormCreateNew from './FormCreateNew';

const CreateNew = () => {

    const [show, setShow] = useState(false);

    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
  

    return (
        <>
          <Button variant="dark" 
                  onClick={handleShow}
          >
            <img 
                src="./plus.png"
                width="25"
                height="25"
                className="d-inline-block align-top"   
                alt=""                 
                >
            </img> 
            Create New
          </Button>
    
          <Modal show={show} onHide={handleClose}>
            <Modal.Header closeButton>
              <Modal.Title>Create a New</Modal.Title>
            </Modal.Header>
            <Modal.Body>

                <FormCreateNew />
                
            </Modal.Body>

          </Modal>
        </>
      );
}

export default CreateNew
