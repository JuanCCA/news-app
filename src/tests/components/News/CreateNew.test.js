import { shallow } from 'enzyme';
import CreateNew from '../../../components/News/CreateNew'
import React from 'react'
import '@testing-library/jest-dom';


describe('Testing <CreateNew />', () => {

    let wrapper = shallow(<CreateNew />)
    const setShow = jest.fn()

    beforeEach( () => {
        jest.clearAllMocks()
        wrapper = shallow(<CreateNew />)
    })
    
    test('should create correctly the component', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('should show modal when button pressed', () => {

        wrapper.find('#buttonModal').simulate('click')

        expect(setShow).toHaveBeenCalled(); //TODO


    })


    

})
