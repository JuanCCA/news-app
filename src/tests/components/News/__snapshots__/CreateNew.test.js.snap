// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CreateNew /> should create correctly the component 1`] = `
<Fragment>
  <Button
    active={false}
    disabled={false}
    id="buttonModal"
    onClick={[Function]}
    variant="dark"
  >
    <img
      alt=""
      className="d-inline-block align-top"
      height="25"
      src="./plus.png"
      width="25"
    />
    Create New
  </Button>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle>
        Create a New
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <FormCreateNew />
    </ModalBody>
  </Modal>
</Fragment>
`;
